File: scrooge-generator/src/main/resources/javagen/readSet.java
TSet _set_{{name}} = _iprot.readSetBegin();
{{#isEnumSet}}
{{name}} = EnumSet.noneOf({{eltType}}.class);
{{/isEnumSet}}
{{^isEnumSet}}
{{name}} = new HashSet<{{eltType}}>();
{{/isEnumSet}}
int _i_{{name}} = 0;
{{eltType}} {{eltName}};
while (_i_{{name}} < _set_{{name}}.size) {
{{#eltReadWriteInfo}}
  {{>readValue}}
{{/eltReadWriteInfo}}
  {{name}}.add({{eltName}});
  _i_{{name}} += 1;
}
_iprot.readSetEnd();

File: scrooge-generator/src/main/resources/cocoagen/readSet.m
int _{{name}}_size;
[inProtocol readSetBeginReturningElementType: NULL size: &_{{name}}_size];
NSUInteger _{{name}}_capacity= _{{name}}_size < 0 ? 0 : (NSUInteger)_{{name}}_size;
NSMutableSet *{{name}}_mutable = [[NSMutableSet alloc] initWithCapacity:_{{name}}_capacity];
for (int _{{name}}_i = 0; _{{name}}_i < _{{name}}_size; ++_{{name}}_i) {
    {{eltType}}{{#eltReadWriteInfo}}{{#isStruct}} *{{/isStruct}}{{/eltReadWriteInfo}} {{eltName}};
{{#eltReadWriteInfo}}
    {{>readValue}}
{{/eltReadWriteInfo}}
    [{{name}}_mutable addObject: {{eltName}}];
}
{{name}} = {{name}}_mutable;
[inProtocol readSetEnd];
